### ISSUE TITLE
`snapshot_not_available()` shouldn't interpolate `message`

### ISSUE THREAD

**Issue by DavisVaughan** [r-lib/testthat#1738]
https://github.com/r-lib/testthat/blob/9cd6e01be008376b1f7f2d8d528d725b87c0d01e/R/snapshot.R#L426-L432

With the following example, you can run the `mutate()` interactively, but you can't run the `expect_snapshot()` call interactively due to `snapshot_not_available()` trying to interpolate `message`, which has `{ ... }` in it, which can't be interpolated

```r
library(testthat)
library(dplyr)

test_that("stuff", {
  df <- tibble(x = 1)

  expect_snapshot(error = TRUE, {
    mutate(df, x = {
      stop("oh no")
    })
  })
})
```

https://user-images.githubusercontent.com/19150088/217931253-a05f084f-77bb-4c20-b8ef-22b058b36f25.mov

One way to avoid this it to do `i = "{message}"` instead, so the interpolation interpolates `message` itself and not its contents

**Comment by gaborcsardi** [comment]
> One way to avoid this it to do i = "{message}" instead, so the interpolation interpolates message itself and not its contents

Yeah, every time you put an arbitrary string into a cli message (or use it in glue in general), the best way is to interpolate it in.

**Comment by hadley** [comment]
Looks like I fixed this incidentally in #1744
